pennylane/qaoa/cycle.py

```diff
@@ -75,3 +77,66 @@ def wires_to_edges(graph: nx.Graph) -> Dict[int, Tuple[int]]:
         Dict[Tuple[int], int]: a mapping from wires to graph edges
     """
     return {i: edge for i, edge in enumerate(graph.edges)}
+
+
+def edge_weight(graph: nx.DiGraph) -> qml.Hamiltonian:
+    r"""Calculates the product of edge weights Hamiltonian.
+
+    The product of weights of a subset of edges in a graph is given by
+
+    .. math:: \prod_{(i, j) \in E} x_{ij} c_{ij}
+
+    where :math:`E` are the edges of the graph, :math:`x_{ij}` is a binary number that selects
+    whether to include the edge :math:`(i, j)` and :math:`c_{ij}` is the corresponding edge weight.
+
+    The product of weights is maximized by equivalently considering
+
+    .. math:: \sum_{(i, j) \in E} x_{ij}\log c_{ij},
+
+    assuming :math:`c_{ij} > 0`.
+
+    This can be restated as a minimization over the following qubit Hamiltonian:
+
+    .. math::
+
+        \sum_{(i, j) \in E} Z_{ij}\log c_{ij}.
+
+    where :math:`Z_{ij}` is a qubit Pauli-Z matrix acting upon the qubit specified by the pair
+    :math:`(i, j)`.
+
+    Args:
+        graph (nx.DiGraph): the graph specifying possible edges
+
+    Returns:
+        qml.Hamiltonian: the product of edge weights Hamiltonian
+
+    Raises:
+        ValueError: if the graph contains parallel edges or self-loops
+        KeyError: if one or more edges do not contain weight data
+    """
+    edges_to_qubits = edges_to_wires(graph)
+    coeffs = []
+    ops = []
+
+    edges_data = graph.edges(data=True)
+    edges = [edge[:2] for edge in graph.edges]
+
+    if len(edges) != len(set(edges)):
+        raise ValueError("Graph contains parallel edges")
```

Comment:
I thnk this can be removed as parallel edges `nx.DiGraphs` according to the [documentation](https://networkx.org/documentation/stable//reference/classes/digraph.html) . A separate class [`nx.MultiDiGraph`](https://networkx.org/documentation/networkx-1.10/reference/classes.multidigraph.html) must be used to support parallel edges

Reply:
Removing this will also resolve the failing `codecov` check